### PR TITLE
Migrate squirrel-update to JS

### PR DESCRIPTION
**Pull request by mnquintana**
### Description of the Change
Blocked on a fix for https://github.com/atom/joanna/issues/5 (in progress)

This PR migrates `squirrel-update.coffee` to modern JS via Babel - unlike some of my past migration attempts, this time around I'm gonna try migrating individual files in their own PRs instead of all bundled into one to try to keep the cost of review and risk of merge conflicts low.

### Possible Drawbacks
This could introduce some bugs into the updater on Windows, but I didn't run into anything from my testing.

👀 @atom/feedback

### CODE REVIEW

File: src/main-process/squirrel-update.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  return Spawner.spawn(updateDotExe, args, callback)
+}
+
+// Add atom and apm to the PATH
+//
+// This is done by adding .cmd shims to the root bin folder in the Atom
+// install directory that point to the newly installed versions inside
+// the versioned app directories.
+function addCommandsToPath (callback) {
+  function installCommands (callback) {
```
Comment (YurySolovyov) [inline src/main-process/squirrel-update.js]: maybe de-nest this one?

File: src/main-process/squirrel-update.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    pathSegments.push(binFolder)
+    const newPathEnv = pathSegments.join(';')
+    return spawnSetx(['Path', newPathEnv], callback)
+  }
+
+  return installCommands((error) => {
+    if (error != null) { return callback(error) }
+
+    return WinPowerShell.getPath((error, pathEnv) => {
+      if (error != null) { return callback(error) }
```
Comment (YurySolovyov) [inline src/main-process/squirrel-update.js]: strict equals here and everywhere?

File: src/main-process/squirrel-update.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+export { spawnUpdate as spawn }
+
+// Is the Update.exe installed with Atom?
+export function existsSync () {
+  return fs.existsSync(updateDotExe)
+}
+
+// Restart Atom using the version pointed to by the atom.cmd shim
+export function restartAtom (app) {
```
Comment (YurySolovyov) [inline src/main-process/squirrel-update.js]: I'm not sure if this is due to some guideline, but I think is usually better to export stuff in one go at the bottom

File: src/main-process/squirrel-update.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  const projectPath = global.atomApplication && global.atomApplication.lastFocusedWindow && global.atomApplication.lastFocusedWindow.projectPath
```
Comment (YurySolovyov) [inline src/main-process/squirrel-update.js]: you might want to break this into multiple lines

File: src/main-process/squirrel-update.js (hunk 3 above)
Comment (BinaryMuse) [inline src/main-process/squirrel-update.js]: For ES2015-style exports I think this is fine. 👍

Comment (YurySolovyov) [inline src/main-process/squirrel-update.js]: Node is doing it too, though it is CJS.
https://github.com/nodejs/node/pull/12654
https://github.com/nodejs/node/pull/12755

This maybe a matter of taste, but I'd say that exports-last style allows you to figure out what's happening in file first, and then just look at what part of that API is exported.

**Comment by mnquintana** [comment]
This PR's waaaay out of date and I'm probably not gonna get to giving this another shot – someone should definitely try converting this to JS though!

**Pull request closed without merging**